File: prak2 php/Memory.php
<?php

class Memory extends Hardware{

    private $frequency;
    private $memorySize;
    private $supportsCuda;

    function __construct($hardware, $fr, $me, $su){
        //konstruktor
        $this->setPrice($hardware->getPrice());
        $this->setIdProduct($hardware->getIdProduct());
        $this->setBrand($hardware->getBrand());
        $this->setModel($hardware->getModel());
        $this->frequency = $fr;
        $this->memorySize = $me;
        $this->supportsCuda = $su;
    }


    //set
    public function setFrequency($fr){
        $this->frequency = $fr;
    }

    public function setMemorySize($me){
        $this->memorySize = $me;
    }

    public function setSupportsCuda($su){
        $this->supportsCuda = $su;
    }

    //get
    public function getFrequency(){
        return $this->frequency;
    }

    public function getMemorySize(){
        return $this->memorySize;
    }

    public function getSupportsCuda(){
        return $this->supportsCuda;
    }


    //methods
    public function tampilkanMemory(){
        echo "Price : ".$this->getPrice()."<br/>";
        echo "idProduct : ".$this->getIdProduct()."<br/>";
        echo "Brand : ".$this->getBrand()."<br/>";
        echo "Model : ".$this->getModel()."<br/>";
        echo "Frequency : ".$this->getFrequency()."<br/>";
        echo "MemorySize : ".$this->getMemorySize()."<br/>";
        echo "SupportsCuda : ".$this->getSupportsCuda()."<br/>";
    }


    function __destruct(){
        //destruktor
    }

}

?>

File: prak2 python/Memory.py
from Hardware import Hardware

class Memory(Hardware):

    def __init__(self, hardware, freq, memory, cuda):
    #konstruktor
        self.price = hardware.getPrice()
        self.idProduct = hardware.getIdProduct()
        self.brand = hardware.getBrand()
        self.model = hardware.getModel()
        self.frequency = freq
        self.MemorySize = memory
        self.supportsCuda = cuda

    #set
    def setFrequency(self, frequency):
        self.frequency = frequency

    def setMemorySize(self, memorySize):
        self.MemorySize = memorySize

    def setSupportsCuda(self, supportsCuda):
        self.supportsCuda = supportsCuda

    #get
    def getFrequency(self):
        return self.frequency

    def getMemorySize(self):
        return self.MemorySize

    def getSupportsCuda(self):
        return self.supportsCuda
    
    #methods
    def tampilkanMemory(self):
        print("Price : " + str(self.getPrice()))
        print("idProduct : " + self.getIdProduct())
        print("Brand : " + self.getBrand())
        print("Model : " + self.getModel())
        print("Frequency : " + self.getFrequency())
        print("MemorySize : " + self.getMemorySize())
        print("SupportsCuda : " + self.getSupportsCuda())

